src/validation.cpp

```diff
@@ -2551,6 +2551,7 @@ bool CChainState::ActivateBestChainStep(CValidationState& state, const CChainPar
             // This is likely a fatal error, but keep the mempool consistent,
             // just in case. Only remove from the mempool in this case.
             UpdateMempoolForReorg(disconnectpool, false);
+            AbortNode(state, "Failed to disconnect block; see debug.log for details");
```

Comment:
In commit "[validation] Crash if disconnecting a block fails" (dafa0b7e1615a4e0985ece5d5f044f14753edcf6)

Probably would be good to have a comment saying that it's better to abort than stay active using a chain that doesn't have the most work (so it's clear that handling the error this way is really intentional and this isn't just fragile code).

Reply:
Done.